src/main/java/frc/robot/subsystems/TrapElvSubsystem.java

```diff
@@ -0,0 +1,340 @@
+      scoringMech =
+          baseMech.append(
+              new MechanismLigament2d("Scoring Arm", 0, 0, 10, new Color8Bit(Color.kAqua)));
+      wristMech =
+          scoringMech.append(
+              new MechanismLigament2d("Wrist Mech", 1.5, -170, 5, new Color8Bit(Color.kRed)));
+
+      m_baseElevatorSim =
+          new ElevatorSim(
+              DCMotor.getNEO(1),
+              TrapElvConstants.elevatorGearRatio,
+              TrapElvConstants.elevatorCariageMass,
+              Units.inchesToMeters(1),
+              TrapElvConstants.elevatorMinHight,
+              TrapElvConstants.elevatorMaxHight,
+              true,
+              0);
+
+      m_scoringElevatorSim =
+          new ElevatorSim(
+              DCMotor.getNEO(1),
+              TrapElvConstants.elevatorGearRatio,
+              TrapElvConstants.elevatorCariageMass,
+              Units.inchesToMeters(1),
+              TrapElvConstants.elevatorMinHight,
+              TrapElvConstants.elevatorMaxHight,
+              true,
+              0);
+
+      TrapElvTab.add("Trap Arm Mech", elvMechanism);
+    }
+  }
+
+  // Boolean Suppliers
+  public BooleanSupplier getSourceBreak() {
+    return () -> sourceBreak.get();
+  }
+
+  public BooleanSupplier getGroundBreak() {
+    return () -> groundBreak.get();
+  }
+
+  public BooleanSupplier getBaseLimit() {
+    return () -> baseLimit.get();
+  }
+
+  public BooleanSupplier getScoringLimit() {
+    return () -> scoringLimit.get();
+  }
+
+  // Commands
+  public Command setRoller(double s) {
+    return run(
+        () -> {
+          rollerMotor.set(s);
+        });
+  }
+
+  public Command stopRoller() {
+    return run(
+        () -> {
+          rollerMotor.stopMotor();
+        });
+  }
+
+  public Command intakeSource() {
+    return new SequentialCommandGroup(
+        setTrapArm(TrapElvState.FROM_SOURCE), setRoller(TrapElvConstants.rollerIntakeSpeed));
+  }
+
+  public Command intakeGround() {
+    return new SequentialCommandGroup(
+        setTrapArm(TrapElvState.FROM_INTAKE), setRoller(-TrapElvConstants.rollerIntakeSpeed));
+  }
+
+  public Command scoreAMP() {
+    return new SequentialCommandGroup(
+        setTrapArm(TrapElvState.AMP_SCORE), setRoller(-TrapElvConstants.rollerScoringSpeed));
+  }
+
+  public Command scoreTrap() {
+    return new SequentialCommandGroup(
+        setTrapArm(TrapElvState.TRAP_SCORE), setRoller(TrapElvConstants.rollerScoringSpeed));
+  }
+
+  public Command stowTrapElv() {
+    return new SequentialCommandGroup(setTrapArm(TrapElvState.STOWED), stopRoller());
+  }
+
+  public Command zeroArm() {
+    return startEnd(
+        () -> {
+          // Command for zering elevator if elevator happens to be not at zero
+          // Runs elevator motors until there limit switches are pressed
+          if (!baseLimit.get()) {
+            baseMotor1.set(.1);
+            baseMotor2.set(.1);
+          } else {
+            baseMotor1.stopMotor();
+            baseMotor2.stopMotor();
```

Comment:
The motors should also be zeroed. 
Because the if the limit wired to the DIO pins as suggested by the use of DigitalInput then the SparkMax won't automatically zero